src/charactersheet/viewmodels/abstract/view-model.js

```diff
@@ -0,0 +1,104 @@
+import {
+    CoreManager,
+    Notifications,
+    Utility
+} from 'charactersheet/utilities';
+import { CardEditActionComponent } from 'charactersheet/components/card-edit-actions';
+import { Clazz } from 'charactersheet/models';
+
+import ko from 'knockout';
+
+/**
+ * AbstractViewModel
+ *
+ * Provides base functionality to view Data instances.
+ *
+ * @property modelName {string} the name of the model being displayed.
+ *
+ * @param containerId {string} The dom id for the container of this component.
+ *
+ * @param flip {function} function for displaying/hiding the view. Used to
+ * 'flip' flip-cards, or expand/collapse rows
+ *
+ * @param show {observable} Whether or not this view is displayed.
+ *
+ * @param existingData {optional observable} Existing data (such as data from a parent).
+ *
+ **/
+
+export class AbstractViewModel {
```

Comment:
I've noticed that a lot of things (items, armor, weapons, etc) `extend AbstractViewModel`, but as far as I can see, they don't use any of the functionality from the parent.
For example, `ItemDetailViewModel` is just a view that displays some fields. Does this VM, and other similar ones, need to extend AbstractViewModel?

Reply:
I've removed AbstractViewModel where is was no longer necessary.